alpine/src/main/scala/io/delta/alpine/internal/data/RowParquetRecordImpl.scala

```diff
@@ -0,0 +1,212 @@
+import io.delta.alpine.types._
+
+private[internal] case class RowParquetRecordImpl(
+    private val record: RowParquetRecord,
+    private val schema: StructType,
+    private val timeZone: TimeZone) extends RowParquetRecordJ {
+
+  private val codecConf = ValueCodecConfiguration(timeZone)
+
+  ///////////////////////////////////////////////////////////////////////////
+  // Public API Methods
+  ///////////////////////////////////////////////////////////////////////////
+
+  override def getInt(fieldName: String): Int = getAs[Int](fieldName)
+
+  override def getLong(fieldName: String): Long = getAs[Long](fieldName)
+
+  override def getByte(fieldName: String): Byte = getAs[Byte](fieldName)
+
+  override def getShort(fieldName: String): Short = getAs[Short](fieldName)
+
+  override def getBoolean(fieldName: String): Boolean = getAs[Boolean](fieldName)
+
+  override def getFloat(fieldName: String): Float = getAs[Float](fieldName)
+
+  override def getDouble(fieldName: String): Double = getAs[Double](fieldName)
+
+  override def getString(fieldName: String): String = getAs[String](fieldName)
+
+  override def getBinary(fieldName: String): Array[Byte] = getAs[Array[Byte]](fieldName)
+
+  override def getBigDecimal(fieldName: String): java.math.BigDecimal =
+    getAs[java.math.BigDecimal](fieldName)
+
+  override def getTimestamp(fieldName: String): Timestamp = getAs[Timestamp](fieldName)
+
+  override def getDate(fieldName: String): Date = getAs[Date](fieldName)
+
+  override def getRecord(fieldName: String): RowParquetRecordJ = getAs[RowParquetRecordJ](fieldName)
+
+  override def getList[T](fieldName: String): util.List[T] = getAs[util.List[T]](fieldName)
+
+  override def getMap[K, V](fieldName: String): util.Map[K, V] = getAs[util.Map[K, V]](fieldName)
+
+  ///////////////////////////////////////////////////////////////////////////
+  // Decoding Helper Methods
+  ///////////////////////////////////////////////////////////////////////////
+
+  private def getAs[T](fieldName: String): T = {
+    val schemaField = schema.get(fieldName);
+    val parquetVal = record.get(fieldName)
+
+    if (null == schemaField) throw DeltaErrors.noFieldFoundInSchema(fieldName, schema)
+
+    if (parquetVal == NullValue && !schemaField.isNullable) {
+      throw DeltaErrors.nullValueFoundForNonNullSchemaField(fieldName, schema)
+    }
+
+    decode(schemaField.getDataType, parquetVal).asInstanceOf[T]
+  }
+
+  private def decode(elemType: DataType, parquetVal: Value): Any = {
+    val elemTypeName = elemType.getTypeName
+    if (primitiveDecodeMap.contains(elemTypeName)) {
+      return primitiveDecodeMap(elemTypeName).decode(parquetVal, codecConf)
+    }
+
+    (elemType, parquetVal) match {
+      case (x: ArrayType, y: ListParquetRecord) => decodeList(x.getElementType, y)
+      case (x: MapType, y: MapParquetRecord) => decodeMap(x.getKeyType, x.getValueType, y)
+      case (x: StructType, y: RowParquetRecord) => RowParquetRecordImpl(y, x, timeZone)
```

Comment:
Could you add an exception for `case _`? This may be useful when we need a bug in this class.